### PR TITLE
[static-build][next][node][redwood] Invoke `installCommand` from Project settings

### PR DESCRIPTION
**Pull request by TooTallNate**
Invokes the `installCommand` instead of yarn/npm if defined in the Project's settings for `@vercel/next`, `@vercel/redwood` and `@vercel/static-build` Runtimes.

Also removes the `--prefer-offline` flag by default for all of the above Runtimes, as well as `@vercel/node`.

### CODE REVIEW

File: packages/now-next/src/index.ts [hunk 1]
```diff
@@ -376,8 +376,18 @@ export const build = async ({
     await writeNpmRc(entryPath, process.env.NPM_AUTH_TOKEN);
   }
 
-  console.log('Installing dependencies...');
-  await runNpmInstall(entryPath, [], spawnOpts, meta);
+  const installTime = Date.now();
```
Comment (styfle) [inline packages/now-next/src/index.ts]: This variable should be in the `else` block right?

File: packages/now-next/src/index.ts [hunk 2]
```diff
@@ -376,8 +376,18 @@ export const build = async ({
...
+  if (typeof installCommand === 'string') {
+    console.log(`Running "install" command: \`${installCommand}\`...`);
+    await execCommand(installCommand, {
+      ...spawnOpts,
+      cwd: entryPath,
+    });
```
Comment (styfle) [inline packages/now-next/src/index.ts]: Let's also pass `prettyCommand: installCommand` here so errors look nice.

You'll probably need to change the type of the execCommand() function to `SpawnOptionsExtended`

File: packages/now-next/src/index.ts (hunk 1 above)
Comment (TooTallNate) [reply packages/now-next/src/index.ts]: Ya, updated.

File: packages/now-node/src/index.ts [hunk 3]
```diff
@@ -104,12 +104,7 @@ async function downloadInstallAndBundle({
   } else {
     const installTime = Date.now();
     console.log('Installing dependencies...');
-    await runNpmInstall(
-      entrypointFsDirname,
-      ['--prefer-offline'],
-      spawnOpts,
-      meta
-    );
+    await runNpmInstall(entrypointFsDirname, [], spawnOpts, meta);
```
Comment (styfle) [inline packages/now-node/src/index.ts:107]: This is missing the `config.installCommand`

File: packages/now-next/src/index.ts (hunk 2 above)
Comment (TooTallNate) [reply packages/now-next/src/index.ts]: Seems that [`execCommand()` already passes `prettyCommand`](https://github.com/vercel/vercel/blob/e2db7c77347fe7698a76df8f31062b9c3b1490b4/packages/now-build-utils/src/fs/run-user-scripts.ts#L110).

File: packages/now-node/src/index.ts (hunk 3 above)
Comment (TooTallNate) [reply packages/now-node/src/index.ts:107]: `installCommand` is for frontend frameworks, not serverless functions.

File: packages/now-static-build/src/index.ts [hunk 4]
```diff
@@ -120,10 +120,10 @@ function getCommand(
     return null;
   }
 
-  const propName = name === 'build' ? 'buildCommand' : 'devCommand';
+  const propName = `${name}Command`;
```
Comment (styfle) [inline packages/now-static-build/src/index.ts]: I think you can avoid the any below if you do 
```suggestion
  const propName = `${name}Command` as typeof keyof config;
```

**Review by styfle — COMMENTED** [review]
Looks good but needs tests

File: packages/now-static-build/src/index.ts (hunk 4 above)
Comment (TooTallNate) [reply packages/now-static-build/src/index.ts]: Nice! That worked for the "config" object, but not for the "framework".

**Comment by TooTallNate** [comment]
Tests added.

**Review by styfle — APPROVED** [review]

**Review by Timer — APPROVED** [review]
LGTM from Next.js angle

**Review by styfle — APPROVED** [review]

**Review by ijjk — APPROVED** [review]

**Comment by TooTallNate** [comment]
E2E tests take too long and are flaky, but you can see in [this run](https://github.com/vercel/vercel/runs/1382722098) that the relevant tests (`now-next/test/fixtures/32-custom-install-command`, `now-static-build/test/fixtures/64-custom-install-command` and `redwood/test/fixtures/16-create-redwood-app`) passed. So merging.

**Pull request merged**